File: rabbitmq/src/test/groovy/io/micronaut/configuration/rabbitmq/docs/producer/ProductClient.java
package io.micronaut.configuration.rabbitmq.docs.producer;

// tag::imports[]
import io.micronaut.configuration.rabbitmq.annotation.Binding;
import io.micronaut.configuration.rabbitmq.annotation.RabbitClient;
import io.micronaut.context.annotation.Requires;
import io.reactivex.Completable;
// end::imports[]

@Requires(property = "spec.name", value = "ProducerAcknowledgeSpec")
// tag::clazz[]
@RabbitClient // <1>
public interface ProductClient {

    @Binding("product") // <2>
    Completable send(byte[] data); // <3>
}
// end::clazz[]

File: docs-examples/example-python/src/test/python/micronaut/rabbitmq/docs/quickstart/ProductClient.py
from abc import ABC, abstractmethod

from micronaut.context.annotation import Requires
# tag::imports[]
from micronaut.rabbitmq.annotation import Binding, RabbitClient
# end::imports[]


@Requires(property="spec.name", value="QuickstartSpec")
# tag::clazz[]
@RabbitClient  # <1>
class ProductClient(ABC):

    @Binding("product")  # <2>
    @abstractmethod
    def send(self, data: bytes) -> None:  # <3>
        ...
# end::clazz[]
